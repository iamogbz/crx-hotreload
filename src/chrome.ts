export const getExtensionInfo = () =>
    new Promise(resolve => chrome.management.getSelf(resolve));

export const getExtensionDirectory = () =>
    new Promise(resolve => chrome.runtime.getPackageDirectoryEntry(resolve));

export const getEntries = (dir: DirectoryEntry) =>
    new Promise(resolve => dir.createReader().readEntries(resolve));

export const reloadCurrentTab = () =>
    new Promise(resolve =>
        chrome.tabs.query({ active: true, currentWindow: true }, resolve),
    ).then(([tab]) => {
        if (tab) chrome.tabs.reload(tab.id);
        chrome.runtime.reload();
    });
