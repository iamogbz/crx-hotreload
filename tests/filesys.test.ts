import * as chromeUtils from "chrome";
import { watchChanges } from "filesys";
import {
    mockDirEntry,
    mockEmptyDirEntry,
    mockFileEntry,
} from "./mocks/filesys";

const logSpy = jest.spyOn(chromeUtils, "log");
const getEntriesSpy = jest.spyOn(chromeUtils, "getEntries");

it("watches for changes to directory", async () => {
    const mockEntries = [mockFileEntry, mockEmptyDirEntry];
    getEntriesSpy
        .mockResolvedValueOnce(mockEntries)
        .mockResolvedValueOnce(mockEntries)
        .mockResolvedValueOnce(mockEntries)
        .mockResolvedValueOnce([mockFileEntry]);
    await new Promise(resolve => watchChanges(mockDirEntry, resolve, 0));
    expect(logSpy.mock.calls).toMatchSnapshot();
});
