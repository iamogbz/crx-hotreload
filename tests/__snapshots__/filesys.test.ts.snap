// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watches for changes to directory 1`] = `
[
  [
    "ðŸ§¨ Watching changes to files in '/mock/full/path'",
  ],
  [
    "ðŸ§¨ Change detected to files in '/mock/full/path'",
  ],
]
`;
